client/src/helpers/storeSearch.ts

```diff
@@ -0,0 +1,340 @@
+import { DEFAULT_NGRAM_LENGTH } from "src/constants";
+import { Store } from "src/interfaces";
+import { Heap } from "src/packages/Heap";
+
+/**
+ * Preprocesses a string s to use as input for calculateEditDistance().
+ *
+ * @param s
+ */
+export const preprocess = (s: string) => {
```

Comment:
add some tests for this

Reply:
Agree, please add tests for the preprocess, try to cover each cases and maybe some composite cases.

Reply:
Added in #242